hi-ml-histopathology/testhisto/testhisto/datasets/test_panda_slides.py

```diff
@@ -0,0 +1,91 @@
+#  -------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  -------------------------------------------------------------------------------------------
+
+from pathlib import Path
+from typing import Any, List
+from unittest import mock
+from unittest.mock import MagicMock
+
+import numpy as np
+import pytest
+import torch
+from pytorch_lightning import seed_everything
+
+from histopathology.configs.classification.DeepSMILESlidesPandaBenchmark import DeepSMILESlidesPandaBenchmark
+from histopathology.utils.naming import SlideKey
+from testhisto.mocks.base_data_generator import MockHistoDataType
+from testhisto.mocks.slides_generator import MockPandaSlidesGenerator, TilesPositioningType
+
+try:
+    from cucim import CuImage  # noqa: F401
+    has_cucim = True
+except:  # noqa: E722
+    has_cucim = False
+
+
+@pytest.mark.skipif(not has_cucim, reason="Test requires CUCIM library")
+@pytest.mark.gpu
+def test_panda_reproducibility(tmp_path: Path) -> None:
+    """Check if subsequent enumerations of the Panda dataset produce identical sequences of tiles."""
+    seed_everything(seed=123, workers=True)
+    tile_size = 28
+    num_tiles = 4
+    wsi_generator = MockPandaSlidesGenerator(
+        dest_data_path=tmp_path,
+        mock_type=MockHistoDataType.FAKE,
+        n_tiles=num_tiles,
+        n_slides=10,
+        n_channels=3,
+        n_levels=3,
+        tile_size=tile_size,
+        background_val=255,
+        tiles_pos_type=TilesPositioningType.RANDOM
+    )
+    wsi_generator.generate_mock_histo_data()
+
+    container = DeepSMILESlidesPandaBenchmark()
+    container.tile_size = tile_size
+    container.max_bag_size = num_tiles
+    container.local_datasets = [tmp_path]
+
+    def test_data_items_are_equal(loader_fn_names: List[str]) -> None:
+        """Creates a new data module from the container, and checks if all the data loaders specified in
+        `loader_fn_names` return the same set of items when enumerated twice."""
+        data_module = container.get_data_module()
+        data_module.dataloader_kwargs = {**data_module.dataloader_kwargs,
+                                         "multiprocessing_context": None,
+                                         "num_workers": 0}
```

Comment:
souldn't we test the case with multi_workers as well? to make sure the changes (in model_trainer) are effective

Reply:
Just using num_workers>0 will not be enough here because we need to test if Lightning does the right thing. Testing the end-to-end training would take the PR effectively onto @mebristo's territory - she has been working on all those tests, but is blocked now. I'll chat with her.

Reply:
Fair enough.